src/java/com/rapleaf/jack/queries/Record.java

```diff
@@ -72,6 +77,74 @@ public Boolean getBoolean(Column column) {
     return value == null ? null : (Boolean)value;
   }
 
+  public <A extends AttributesWithId, M extends ModelWithId> A getAttributes(Table<A, M> tableType) {
+    String tableName = tableType.getAlias();
+    Constructor<A> constructor;
+    try {
+      constructor = tableType.getAttributesType().getConstructor(Long.TYPE);
+    } catch (NoSuchMethodException e) {
+      throw new RuntimeException(e);
+    }
+
+    if (constructor == null || tableName == null) {
+      throw new RuntimeException("No columns from Table " + tableName + " are included in this record");
+    }
+
+    Long id = null;
+    Map<Enum, Object> fieldMap = Maps.newHashMap();
+    for (Map.Entry<Column, Object> entry : columns.entrySet()) {
+      Column column = entry.getKey();
+      if (column.getTable().equals(tableName)) {
+        if (column.getField() != null) {
+          fieldMap.put(column.getField(), entry.getValue());
+        } else if (id == null) {
+          id = ((Number)(entry.getValue())).longValue();
+        } else {
+          throw new RuntimeException("The record contains multiple IDs for Table " + tableName);
```

Comment:
This problem is actually fixed in this pull request. This check, `column.getTable().equals(tableName)`, filters the fields by the table name to prevent it. In the test case, I did test it with a join query from two tables:
```
Record record = db.createQuery()
    .from(User.TBL)
    .innerJoin(Post.TBL).on(Post.USER_ID.equalTo(User.ID))
    ...
```

The `else` branch could only happen if someone manually constructs a `Record` with two `ID` fields from the same table, which is highly unlikely.